testing/e2e/agent_reconnection_test.go

```diff
@@ -7,25 +7,32 @@ import (
 	"time"
 
 	"github.com/sensu/sensu-go/types"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestAgentReconnection(t *testing.T) {
 	t.Parallel()
 
-	// Start the backend
-	backend, cleanup := newBackend(t)
+	backend, cleanup, err := newBackendProcess(40005, 40006, 40007, 40008, 40009)
```

Comment:
Using explicit port numbers for testing will really help when tracing port issues down to the originating test. It does mean that we would have to manually avoid port collisions though. I wonder if there's a better way to coordinate this.

Reply:
I've tried a few approaches in the past. It's pretty tricky to do right, because the only way you can "reserve" a port is by binding to it.

Since the backend takes port numbers, and not listeners, there is no way to just ask the OS for a free port.

This is one of the reasons I was considering docker for the e2e tests.